lib/web_ui/lib/src/engine/platform_views/message_handler.dart

```diff
@@ -73,7 +73,8 @@ class PlatformViewMessageHandler {
     if (!_contentManager.knowsViewType(viewType)) {
       callback(_codec.encodeErrorEnvelope(
         code: 'unregistered_view_type',
-        message: 'trying to create a view with an unregistered type',
+        message: 'trying to create a view with an unregistered type. '
+            'Have you called GeneratedPluginRegistrant.register?',
```

Comment:
On web I think this is `registerPlugins()`? @ditman can you confirm?

Reply:
@jmagman thanks for pinging me here!

In the web, well formed plugins are registered automatically by the flutter tool (it generates a plugin_registrant.dart file, which wires all the plugins up). We have [one exception](https://pub.dev/packages/webview_flutter_web#usage) that is being [fixed](https://github.com/flutter/flutter/issues/94051).

The main cause of this error on the web is not forgetting to register a plugin, IMO. The problem happens when a web app attempts to use an [`HtmlElementView` widget](https://api.flutter.dev/flutter/widgets/HtmlElementView-class.html) with a `viewType` that hasn't had its "view factory" registered previously. Users who create their own `HtmlElementView` widgets are supposed to register them through: `ui.platformViewRegistry.registerViewFactory`.

It can be the case that the registerViewFactory call is made by a plugin that provides its own widget (like [google_maps_flutter_web](https://github.com/flutter/plugins/blob/7686be7ecbe1cc049423715b5e2d897cb63a43c1/packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart#L100-L111), or [video_player_web](https://github.com/flutter/plugins/blob/7686be7ecbe1cc049423715b5e2d897cb63a43c1/packages/video_player/video_player_web/lib/video_player_web.dart#L198-L199)), but it can also happen that a programmer is attempting to use an `HtmlElementView` defined on your app but forgot to call `registerViewFactory`.

I'm not sure the change above makes the problem easier to resolve (at least on the web platform?). I wish we had a good documentation link that we could point users to, but I can't find anything :/

Reply:
@ditman @jmagman

I can either change the text for web to something like: "Have you called `ui.platformViewRegistry.registerViewFactory`?", or I can undo the change here, leave on this PR only the error message for iOS, and then create another issue for web.

What do you think?

Reply:
@dacianf I think that @cyanglaz has a good suggestion in his comment. For the web, I don't mind if you leave this as is, or you update the `trying to create a view with an unregistered type.` part to something similar to what Chris suggested:

```
'A HtmlElementView widget is trying to create a platform view with an unregistered type: <some type>.'
'If you are the author of the PlatformView, make sure `registerViewFactory` is invoked.'
```

That message is much more descriptive than the current error, I think?

(Also, it'd be super helpful if you could create a new issue to improve the platform web documentation in this regard, so we can link to something like what was suggested for the iOS message 😅)

Reply:
Thanks again for pinging me @jmagman, and for the updates @dacianf. I'm marking this bit of the conversation as _resolved_!